src/cfg/mod.rs

```diff
@@ -24,25 +26,59 @@ pub(crate) type NodeSet = <StableDiGraph<BasicBlock, Branch> as Visitable>::Map;
 mod tests;
 
 pub(crate) mod structured;
+pub(crate) mod to_bril;
 pub(crate) mod to_structured;
 
 /// Convert a program to a cfg.
 /// Loops over all the functions, translating individually.
-pub(crate) fn program_to_cfg(program: &Program) -> CfgProgram {
+pub(crate) fn program_to_cfg(program: &Program) -> SimpleCfgProgram {
     let mut functions = Vec::new();
     for func in &program.functions {
-        let cfg = to_cfg(func);
+        let cfg = function_to_cfg(func);
         functions.push(cfg);
     }
     CfgProgram { functions }
 }
 
 #[derive(Clone)]
-pub struct CfgProgram {
-    pub functions: Vec<Cfg>,
+pub struct CfgProgram<CfgType> {
+    pub functions: Vec<CfgFunction<CfgType>>,
+}
+
+/// Simple programs only branch on booleans defined in bril
+pub type SimpleCfgProgram = CfgProgram<Simple>;
+/// Switch programs can also branch on values defined in annotations
+pub type SwitchCfgProgram = CfgProgram<Switch>;
+
+impl SimpleCfgProgram {
+    // Convert a simple program to a switch program
+    // trivial, since simple programs are a subset of switch programs
+    pub fn convert_to_switch(self) -> SwitchCfgProgram {
+        SwitchCfgProgram {
+            functions: self
+                .functions
+                .into_iter()
+                .map(|f| f.convert_to_switch())
+                .collect(),
+        }
+    }
 }
 
-impl CfgProgram {
+impl SimpleCfgFunction {
+    pub fn convert_to_switch(self) -> SwitchCfgFunction {
+        SwitchCfgFunction {
+            args: self.args,
+            graph: self.graph,
+            entry: self.entry,
+            exit: self.exit,
+            name: self.name,
+            return_ty: self.return_ty,
+            phantom: PhantomData,
```

Comment:
Needs this PhantomData thingy to keep the Simple apart from the Switch. I'm not sure I love this, but it's kind of a cool trick. This way we get to share a lot of code, and we just don't use annotations for the simple kind.

Reply:
Yeah, that's right. Thanks for the example code, I was considering something similar and I like the guarantee that we have certain annotations. 

I think for now let's leave it as is. I think there are several differences I'd like to enforce between the two but converting between them is cumbersome. Can revisit this question in the future